Add Archive and Searchable Post Statuses

funkSoRubber opened the pull request:
Two custom post status were created; archive and searchable. Archived funds will not be searchable or appear on the priorities list, while searchable funds will only show in the search box. A script is used to insert the two statuses in the status selector drop down only when a user is on a fund post type. 

The WordPress API only lets you query by status when authenticated, so a custom API endpoint was created on the server side to search by status. The 'post_status' => 'any' property in the API endpoint includes statuses with exclude_from_search (property set on the post status) equal to true.

Since we use the WordPress API for gathering priorities, only funds in the published status will appear, no code changes necessary.

blairlierman commented:
@jeremyfelt or @philcable #reviewmerge please

philcable commented on includes/class-custom-post-status.php:
```diff
@@ -0,0 +1,77 @@
+		));
+	}
+
+	function append_post_status_list() {
```
I know it's kind of a pain for such a small amount of JavaScript, but I'd feel better about this if it was a separate file (the variables can be handled via `wp_localize_script` when you enqueue the script).

This is mostly personal preference, though :)

philcable commented on includes/wsuwp-shortcode-fundselector.js:
```diff
@@ -7,31 +7,21 @@ jQuery(document).ready(function($) {
         {
 			source: function( request, response ) {
 				$("#fundSearch").addClass("loading");
-				$.getJSON( wpData.request_url_base + 'idonate_fund', 
```
This would need to be modified slightly if the `filter[s]` parameter suggestion above is sufficient.

funkSoRubber commented on includes/class-wsuwp-plugin-idonate-restapi.php:
```diff
@@ -119,4 +120,35 @@ function wsuf_fundselector_funds_get_fund_by_des_id_rest( $data ) {
+	*
+	* @since 1.1.2
+	**/
+	function wsuf_fundselector_register_endpoint_fund_search() {
```
@philcable Is filter[s] an actual query parameter we can use? I tried to find some documentation on how to use it, but couldn't get any useful results. I had tried using the default endpoint with ?status=searchable, but for the default endpoint I was getting errors that made me think that the request had to be authenticated first. The custom endpoint let us workaround the authentication issue.

funkSoRubber commented:
@blairlierman I'm pretty sure we introduced the issue here when I modified the search based off of statuses. I think we need to decode the search term to search it against the funds in WordPress. If you try /wp-json/idonate_fundselector/v1/search/computer%20science it doesn't show anything
